lib/core/src/animation/physics.rs

```diff
@@ -0,0 +1,308 @@
+    }
+
+    pub fn mod_spring<F:FnOnce(&mut SpringProperties)>(&mut self, f:F) {
+        f(&mut self.data.borrow_mut().spring)
+    }
+
+    pub fn mod_drag<F:FnOnce(&mut DragProperties)>(&mut self, f:F) {
+        f(&mut self.data.borrow_mut().drag)
+    }
+}
+
+
+
+// ========================
+// === SimulationObject ===
+// ========================
+
+pub trait SimulationObject = HasPosition + 'static;
+
+
+
+// ========================
+// === PhysicsSimulator ===
+// ========================
+
+/// A 60 steps per second physics simulator used to simulate `Properties`.
+pub struct PhysicsSimulator {
+    _fixed_step_animator : FixedStepAnimator
+}
+
+impl PhysicsSimulator {
+    /// Simulates `Properties` on `object`.
+    pub fn new<T:SimulationObject>(mut object:T, mut properties:PhysicsProperties) -> Self {
+        properties.mod_kinematics(|kinematics| { kinematics.set_position(object.position()); });
+
+        let steps_per_second = 60.0;
+        let step_time        = 1.0 / steps_per_second;
+        let current_position = Rc::new(RefCell::new(object.position()));
+        let next_position    = Rc::new(RefCell::new(simulate(&mut properties, step_time)));
+
+        let current_position_clone  = current_position.clone();
+        let next_position_clone     = next_position.clone();
+        let mut continuous_animator = ContinuousTimeAnimator::new(move |time| {
+            let current_position = *current_position_clone.borrow();
+            let next_position    = *next_position_clone.borrow();
+            let transition       = time / step_time / 1000.0;
+            object.set_position(linear_interpolation(current_position, next_position, transition));
+        });
+
+        let _fixed_step_animator   = FixedStepAnimator::new(steps_per_second, move |_| {
+            continuous_animator.set_time(0.0);
+            let mut next_position          = next_position.borrow_mut();
+            *current_position.borrow_mut() = *next_position;
+            *next_position                 = simulate(&mut properties, step_time);
+        });
+
+        Self { _fixed_step_animator }
+    }
+}
+
+/// Simulate the `KinematicProperties`.
+fn simulate_kinematics(kinematics:&mut KinematicsProperties, force:&Vector3<f32>, dt:f32) {
+    kinematics.set_acceleration(force / kinematics.mass);
+    kinematics.set_velocity(kinematics.velocity() + kinematics.acceleration() * dt);
+    kinematics.set_position(kinematics.position() + kinematics.velocity()     * dt);
+}
+
+/// Runs a simulation step.
+fn simulate(properties:&mut PhysicsProperties, delta_time:f32) -> Vector3<f32> {
+    let spring        = properties.spring();
+    let drag          = properties.drag();
+    let mut net_force = zero();
+    properties.mod_kinematics(|mut kinematics| {
+        net_force += spring.force(&kinematics);
+        net_force += drag.force(&kinematics);
+        simulate_kinematics(&mut kinematics, &net_force, delta_time);
```

Comment:
@wdanilo see this section. This code could be far more extensible and it could be refactored to

```rust
struct Properties {
    forces : Vec<Box<dyn PhysicsForce>>,
    kinematics : KinematicsProperties
}

for f in properties.forces {
    net_force += f.force();
}
```

But I can't do it because only `PhysicsProperties` implements the shared mutability pattern we wanted to get rid of to avoid RefCell borrowing overhead.

If each property implemented the same design pattern, I would be able to remove `PhysicsKinematics` parameter from

```rust
fn force(&self, kinematics:&PhysicsKinematics)
```

And share it in `SpringProperties`, because it needs to know the body position.

And I would also be able to mutate `SpringProperties` without getting its reference from

```rust
struct PhysicsProperties {
    pub fn mod_spring<F:FnOnce(&mut SpringProperties)>(&mut self, f:F) {
        f(&mut self.data.borrow_mut().spring)
    }
}
```

In fact, this interface would be deprecated.

I didn't want to make this move without consulting for better time management. I am moving back to the `Camera Animation System` PR now.